rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1034,28 +1071,35 @@ private void allowFederationElectionAgain() {
 
     private boolean svpIsOngoing() {
         return federationSupport.getProposedFederation()
-            .map(Federation::getCreationBlockNumber)
-            .map(proposedFederationCreationBlockNumber ->
-                proposedFederationCreationBlockNumber + bridgeConstants.getFederationConstants().getValidationPeriodDurationInBlocks())
-            .filter(validationPeriodEndBlock -> rskExecutionBlock.getNumber() <= validationPeriodEndBlock)
+            .filter(this::validationPeriodIsOngoing)
             .isPresent();
     }
 
-    protected void processSvpFundTransactionUnsigned(Transaction rskTx) throws IOException, InsufficientMoneyException {
-        Optional<Federation> proposedFederationOpt = federationSupport.getProposedFederation();
-        if (proposedFederationOpt.isEmpty()) {
-            String message = "Proposed federation should be present when processing SVP fund transaction.";
-            logger.warn(message);
-            throw new IllegalStateException(message);
-        }
-        Federation proposedFederation = proposedFederationOpt.get();
+    private boolean validationPeriodIsOngoing(Federation proposedFederation) {
+        long validationPeriodEndBlock = proposedFederation.getCreationBlockNumber() +
+            bridgeConstants.getFederationConstants().getValidationPeriodDurationInBlocks();
 
-        Coin spendableValueFromProposedFederation = bridgeConstants.getSpendableValueFromProposedFederation();
-        BtcTransaction svpFundTransactionUnsigned = createSvpFundTransaction(proposedFederation, spendableValueFromProposedFederation);
+        return rskExecutionBlock.getNumber() <= validationPeriodEndBlock;
+    }
 
-        provider.setSvpFundTxHashUnsigned(svpFundTransactionUnsigned.getHash());
-        PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = provider.getPegoutsWaitingForConfirmations();
-        settleReleaseRequest(pegoutsWaitingForConfirmations, svpFundTransactionUnsigned, rskTx.getHash(), spendableValueFromProposedFederation);
+    private void processSvpFundTransactionUnsigned(Keccak256 rskTxHash, Federation proposedFederation) {
+        Coin spendableValueFromProposedFederation = bridgeConstants.getSpendableValueFromProposedFederation();
+        try {
+            BtcTransaction svpFundTransactionUnsigned = createSvpFundTransaction(proposedFederation, spendableValueFromProposedFederation);
+            provider.setSvpFundTxHashUnsigned(svpFundTransactionUnsigned.getHash());
+            PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = provider.getPegoutsWaitingForConfirmations();
+            settleReleaseRequest(pegoutsWaitingForConfirmations, svpFundTransactionUnsigned, rskTxHash, spendableValueFromProposedFederation);
+        } catch (InsufficientMoneyException e) {
+            logger.error(
+                "[processSvpFundTransactionUnsigned] Insufficient funds for creating the fund transaction. Error message: {}",
+                e.getMessage()
```

Comment:
I recommend logging the entire Throwable not just the message. That way we can trace what happened. 
```suggestion
                "[processSvpFundTransactionUnsigned] Insufficient funds for creating the fund transaction",
                e
```

Reply:
That seems strange, do you remember what it said?

Reply:
not really

Reply:
Just noting here, having the entire stacktrace enable us to know what exactly happened which will gives us a better insight to what occurred during runtime

Reply:
I think you need to get rid if the `{}`
```
log("an error ocurred", e)
```
I think, not 100% sure

Reply:
that was it. Thanks guys :)